src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -50,10 +51,8 @@ public SolutionStateChecksums(params object[] children) : base(WellKnownSynchron
                 result[Projects.Checksum] = Projects;
             }
 
-            foreach (var kv in state.ProjectStates)
+            foreach (var projectState in state.ProjectStates.Values)
```

Comment:
💡 `ImmutableDictionary<TKey, TValue>` has a struct enumerator, but `ImmutableDictionary<TKey, TValue>.Values` does not. The former is more efficient:

```suggestion
            foreach (var (_, projectState) in state.ProjectStates)
```

See also #19126

Reply:
Hmm, that's very subtle. Perhaps we should have an analyzer for this.